### ISSUE TITLE
feat: production deploys from tagged contract versions

### ISSUE THREAD

**Issue by anna-carroll** [nomad-xyz/monorepo#325]
### Package

deploy

### Describe the feature you would like

need to design how we will handle versioning between `contracts-core`, `contracts-bridge`, `contracts-router`, and `deploy` packages, for each use case where they are needed

As the contract code + interfaces change, the deploy script must also change. 

**Use cases:** 
- production deploys: deploy from tagged contract version
- development + staging deploys: deploy from tagged contract version?
- local environment testing: default = local workspace version, possibly able to deploy from tagged versions too 

We cannot just update the `package.json` of deploy package to suit the use case, as we _also_ need to use the right version of the deploy package itself

**Comment by anna-carroll** [comment]
> ya, for any given commit, CI should guarantee that the version of deploy in that commit is compatible with its dependency. So when we tag a contract release, we need to bump the deploy script dependency version. If we always do that correctly, we should keep things in sync I guess? 🤔

hmmmm wdyt @Imti

**Comment by anna-carroll** [comment]
@prestwich logistically how would we check whether version of `deploy` is compatible with version of `contracts-*` deps? there may be multiple versions on each side that are compatible with each other

**Comment by anna-carroll** [comment]
in other words, it's not a 1:1 mapping between `deploy` and `contracts-*` package versions 
(we may tag a new contracts release which doesn't change the deploy script; or we may update the deploy script without changing the contracts)

**Comment by anna-carroll** [comment]
> I'm saying that CI already will fail to build if `deploy` specifies a `contracts-*` version that deploy's code is type-system incompatible with

ah, yes. this is true. 

but.. how do we deploy from past versions? i guess just check out that version in a branch.

**Comment by prestwich** [comment]
> but.. how do we deploy from past versions? i guess just check out that version in a branch.

yes

**Comment by Imti** [comment]
I agree with @prestwich, basically following the convention for npm packages releases. Let me know if you need help setting this up.

**Comment by odyslam** [comment]
I think we have fixed this @anna-carroll ?
